packages/flutter/lib/src/widgets/container.dart

```diff
@@ -381,16 +370,29 @@ class Container extends StatelessWidget {
 
   @override
   Widget build(BuildContext context) {
+    assert(margin == null || margin!.isNonNegative);
+    assert(padding == null || padding!.isNonNegative);
+    assert(decoration == null || decoration!.debugAssertIsValid());
+    assert(constraints == null || constraints!.debugAssertIsValid());
+    assert(decoration != null || clipBehavior == Clip.none);
+    assert(
+      color == null || decoration == null,
+      'Cannot provide both a color and a decoration\n'
+      'To provide both, use "decoration: BoxDecoration(color: color)".',
+    );
```

Comment:
This will delay these exceptions to build time instead of widget construction time, which will make things harder to debug.